dcpy/connectors/edm/recipes.py

```diff
@@ -28,29 +28,27 @@
 LOGGING_TABLE_NAME = "metadata_logging"
 
 
-def archive_raw_dataset(ingest_config: ingest.Config, file_path: Path):
+def _archive_dataset(config: ingest.Config, file_path: Path, s3_path: Path):
     with TemporaryDirectory() as tmp_dir:
         tmp_dir_path = Path(tmp_dir)
-        shutil.copy(file_path, tmp_dir)
-        # config = library.Config(dataset=ingest_config, execution_details=None)
-
+        shutil.copy(file_path, tmp_dir_path)
         with open(tmp_dir_path / "config.json", "w") as f:
-            f.write(json.dumps(ingest_config.model_dump(), indent=4))
+            f.write(json.dumps(config.model_dump(), indent=4))
         s3.upload_folder(
             BUCKET,
             tmp_dir_path,
-            ingest_config.raw_dataset_key.s3_path(RAW_FOLDER),
-            acl=ingest_config.acl,
+            s3_path,
+            acl=config.acl,
             contents_only=True,
         )
 
 
-def archive_dataset():
-    ### this could either take a config object and path to a file, and archive them both in a folder together,
-    ### or take path to folder that takes dumped config as well as file output and loads whole folder
-    ### likely the latter in case of multiple output formats. Not necessarily something we want to be doing long term
-    ### but for flexibility for now it might make sense
-    raise NotImplemented
+def archive_raw_dataset(config: ingest.Config, file_path: Path):
+    _archive_dataset(config, file_path, config.raw_dataset_key.s3_path(RAW_FOLDER))
+
+
+def archive_dataset(config: ingest.Config, file_path: Path):
```

Comment:
Could you please explain the need for `_archive_dataset()`, `archive_raw_dataset()`, and `archive_dataset()`? Seems somewhat redundant...

Reply:
Yeah that's fair. When I started, I wrote them as separate functions before realizin how much code was shared.

I think to some extent, it's nice that when archiving from `ingest` code, you don't have to worry about things like s3 prefixes, and just have atomic operations that make sense in the context of the "run". At the beginning, simply a command to archive the raw dataset, at the end, a command to archive the processed dataset, just based on the config and file.

Main difference in functionality is that raw datasets go `raw_datasets/{timestamp}/{raw_filename}` whereas processed are `datasets/{version}/{dataset_name}.parquet`.

Could definitely do it with a boolean value instead (`raw: bool = False`), or just pass the s3 prefix. I do prefer not the latter personally, just a tad.

Reply:
I see, thanks for the description. I don't have a strong a preference for either having the 2 separate functions for `ingest` archiving or 1 function with an input param, so I defer to you. It would be helpful to add function docstring(s) regardless of the choice.